### ISSUE TITLE
OpenSSL-3.0.0-alpha2 MSYS2/MinGW build failure

### ISSUE THREAD

**Issue by MSP-Greg** [openssl/openssl#11937]
I've been intermittently building MSYS2/MinGW OpenSSL since 1.0.2.

With 3.0.0-alpha2, I always have the following build error:
```
perl "-I." -Mconfigdata "../openssl-3.0.0-alpha2/util/dofile.pl" "-oMakefile" ../openssl-3.0.0-alpha2/include/openssl/opensslv.h.in > include/openssl/opensslv.h
perl "-I." -Mconfigdata "../openssl-3.0.0-alpha2/util/dofile.pl" "-oMakefile" ../openssl-3.0.0-alpha2/test/provider_internal_test.cnf.in > test/provider_internal_test.cnf
make: *** No rule to make target 'doc/man/man1/CA.pl.1.exe', needed by 'build_man_docs'.  Stop.
```
MSYS2 gcc is 10.1.0, perl 5.30, Text::Template 1.58

Any ideas?  Not much of a c or perl type...

[configdata.pm_dump.txt](https://github.com/openssl/openssl/files/4673844/configdata.pm_dump.txt)

**Comment by mattcaswell** [comment]
@levitte?

**Comment by frustreated** [comment]
> I've been intermittently building MSYS2/MinGW OpenSSL since 1.0.2.
> 
> With 3.0.0-alpha2, I always have the following build error:
> 
> ```
> perl "-I." -Mconfigdata "../openssl-3.0.0-alpha2/util/dofile.pl" "-oMakefile" ../openssl-3.0.0-alpha2/include/openssl/opensslv.h.in > include/openssl/opensslv.h
> perl "-I." -Mconfigdata "../openssl-3.0.0-alpha2/util/dofile.pl" "-oMakefile" ../openssl-3.0.0-alpha2/test/provider_internal_test.cnf.in > test/provider_internal_test.cnf
> make: *** No rule to make target 'doc/man/man1/CA.pl.1.exe', needed by 'build_man_docs'.  Stop.
> ```
> 
> MSYS2 gcc is 10.1.0, perl 5.30, Text::Template 1.58
> 
> Any ideas? Not much of a c or perl type...
> 
> [configdata.pm_dump.txt](https://github.com/openssl/openssl/files/4673844/configdata.pm_dump.txt)

hello and i also encounterd this , any idears to solved it?

**Comment by levitte** [comment]
Fixed in #12374

**Comment by MSP-Greg** [comment]
@levitte

Thank you for the fix.  Using MSYS2/MinGW gcc 10.1.0 and alpha4, it runs considerably longer, but now stops with a error of:
```
ar r libcrypto.a crypto/aes/libcrypto-lib-aes-x86_64.obj ...
make[1]: ar: Argument list too long
```

The actual ar line is about 37.8 k long.
